### PR TITLE
Fix:Adds a check for empty search parameters

### LINKED ISSUE #2433: Clear Filter gets highlighted on No Param Passes
**Describe the bug**

When there are no parameters in search bar on startup page and After clicking search , Clear Filter gets highlighted .

**To Reproduce**

1. Go to 'Start-Up Page'
2. Click on 'Search'

**Expected behavior**

Unless some parameter is passed ,Clear-Filter should not get highlighted.

**Screenshots**

Screenshot 2019-03-31 at 5 15 44 AM

**Desktop (please complete the following information):**
 - OS: Mac OS Mojave
 - Browser: Chrome
 - Version: 73.0.3

### LINKED ISSUE #3547: Quick Filter triggers a Get Request with empty values
**Describe the bug**

When the search button on the homepage gets clicked with empty search event and search location,a request is made to the server.
**To Reproduce**
Mouse focus on the event input and location input tabs without typing anything,click the search button and view the search bar
**Expected behaviour**
No search should be triggered with empty event names and Location.

**Screenshots**
Screenshot from 2019-10-05 16-46-54

### PR DESCRIPTION
**Pull request by Batey96**
Fixes #3547
Fixes #2433

#### Short description of what this resolves:
This PR stops the Get request to server with empty search parameters
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Batey96** [comment]
> Now `Clear Filter` gets highlighted without even applying filters.

You mean with these commits @kushthedude ?

**Comment by kushthedude** [comment]
Yes

**Comment by Batey96** [comment]
@kushthedude yes,it makes more sense to clear the filters even if filters are not applied,that is to say clears the present inputted values before submit..It as well clears the filters when they are applied.You may correct me more on this.

**Comment by kushthedude** [comment]
Date filters are not working.

**Comment by Batey96** [comment]
View with all Request filters(name,location and date) working as before but with limitations

Oct 8, 2019 2_38 AM

**Comment by kushthedude** [comment]
That's what I am saying, check your GIF, Search query using `Date Filters` is not working .

**Comment by kushthedude** [comment]
Show me the GIF for a working `Date Query` only

**Comment by Batey96** [comment]
@kushthedude code has been updated.Date filter working which also highlights the clear filter when triggered
Oct 8, 2019 4_35 AM

File: app/components/quick-filter.js [hunk 1]
```diff
@@ -61,25 +63,38 @@ export default Component.extend({
...
       this.setDateFilter();
       this.set('location', this.dummyLocation);
       this.set('eventName', this.dummyName);
       this.set('disableClear', true);
     }
+  },
+
+  dummiessFull() {
+    if (this.dummyLocation.length === 0 && this.dummyName.length === 0) {
+      return 0;
```
Comment (kushthedude) [inline app/components/quick-filter.js]: Make it false.

File: app/controllers/index.js [hunk 2]
```diff
@@ -6,8 +6,8 @@ export default Controller.extend({
   queryParams : ['event_name', 'start_date', 'end_date', 'location'],
   start_date  : null,
   end_date    : null,
-  location    : null,
-  name        : null,
+  location    : '',
+  event_name  : '',
```
Comment (kushthedude) [inline app/controllers/index.js]: dont change the variable name

File: app/components/quick-filter.js (hunk 1 above)
Comment (Batey96) [reply app/components/quick-filter.js]: Done

Comment (Batey96) [reply app/components/quick-filter.js]: @kushthedude all resolved with new commits..

**Comment by iamareebjamal** [comment]
Merging branch dismisses the review. Don't do it

**Comment by iamareebjamal** [comment]
@kushthedude @prateekj117 Test and review

**Comment by iamareebjamal** [comment]
You did it again

**Comment by prateekj117** [comment]
@Batey96 No need for a new PR, just rebase your branch with `development`.

**Comment by Batey96** [comment]
@kushthedude @iamareebjamal @prateekj117  i am on it,will do some googling on github usage inorder to correct my mess.thanks

**Comment by Batey96** [comment]
i have messed up even my dev branch..i will redo a clone from scratch and copy paste my old files to new clone

**Comment by kushthedude** [comment]
PR is orphan. Closing hence.

**Pull request closed without merging**